lib/internal/http2/compat.js

```diff
@@ -428,20 +456,31 @@ class Http2ServerResponse extends Stream {
     stream.respond(headers);
   }
 
+  createPushResponse(headers, options, callback) {
+    var stream = this[kStream];
+    if (!stream) {
+      callback(new Error('HTTP/2 Stream has been closed'));
+      return;
+    }
+    return stream.pushStream(headers, options, callback);
+  }
+
   [kBeginSend]() {
     var state = this[kState];
     var stream = this[kStream];
     if (!state.headersSent) {
       state.headersSent = true;
-      const headers = llistToHeaders(this[kHeaders]);
+      const headers = this[kHeaders] || Object.create(null);
```

Comment:
yes, this is exactly why a linked list was used. I'll have to dig in a bit more to see what is best here.